Improve UX of drive letters (3.2)

RandomShaper opened the pull request:
Version of #36069 for 3.2.

---
**This code is generously donated by IMVU,**

akien-mga commented on core/bind/core_bind.cpp:
```diff
@@ -2462,7 +2462,7 @@ void _Directory::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("get_current_drive"), &_Directory::get_current_drive);
 	ClassDB::bind_method(D_METHOD("change_dir", "todir"), &_Directory::change_dir);
-	ClassDB::bind_method(D_METHOD("get_current_dir"), &_Directory::get_current_dir);
+	ClassDB::bind_method(D_METHOD("get_current_dir", "include_drive"), &_Directory::get_current_dir, DEFVAL(true));
```
I wonder if this doesn't break compat for the GDNative API?

RandomShaper replied:
I don't think so because it's not a core core class (repetition intended) and the parameter is optional. However, to be honest, I'm in doubt. I need some help to confirm.

RandomShaper replied:
Truth is it breaks it indeed. I'll submit an update.

RandomShaper committed:
Improve UX of drive letters

Namely, move the drive dropdown to just the left of the path text box and don't include the former
in the latter.

This improves the UX on Windows.

In the UNIX case, since its concept of drives is (ab)used to provide shortcuts to useful paths, its
dropdown is kept at the original location.

akien-mga commented:
Thanks!
